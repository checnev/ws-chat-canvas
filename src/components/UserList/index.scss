.userList {
  list-style: none;
  padding-left: 0;
  margin: 0;

  li {
    margin-bottom: 10px;
  }
}
