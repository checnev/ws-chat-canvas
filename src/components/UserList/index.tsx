import React from 'react';
import cn from 'utils/cn';
import clsx from 'clsx';
import UserInfo from 'components/UserInfo';
import './index.scss';
import type { IUser } from 'lib/user/User';

interface UserListProps {
  users: IUser[];
  className?: string;
}

class UserList extends React.Component<UserListProps> {
  render() {
    const userList = cn('userList');

    return (
      <ul className={clsx(this.props.className, userList())}>
        {this.props.users.map((user) => (
          <li key={user.name}>
            <UserInfo user={user} />
          </li>
        ))}
      </ul>
    );
  }
}

export default UserList;
