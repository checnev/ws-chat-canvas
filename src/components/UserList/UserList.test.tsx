import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import UserList from 'components/UserList';

let container: any = null;

beforeEach(() => {
  container = document.createElement('div');
  document.body.append(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('should render UserList', () => {
  const users = [
    {
      name: 'Naruto',
      color: 'green',
    },
    {
      name: 'Sasuke',
      color: 'orange',
    },
  ];

  act(() => {
    render(<UserList users={users} />, container);
  });

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
});
