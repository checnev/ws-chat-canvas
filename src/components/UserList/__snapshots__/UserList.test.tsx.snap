// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render UserList 1`] = `
"<ul class=\\"userList\\">
  <li>
    <div class=\\"userInfo\\">
      <div class=\\"userInfo__avatar\\" style=\\"background-color: green;\\">N</div><span class=\\"userInfo__username\\">Naruto</span>
    </div>
  </li>
  <li>
    <div class=\\"userInfo\\">
      <div class=\\"userInfo__avatar\\" style=\\"background-color: orange;\\">S</div><span class=\\"userInfo__username\\">Sasuke</span>
    </div>
  </li>
</ul>"
`;
